lib/fetch/dataURL.js

```diff
@@ -126,7 +126,13 @@ function URLSerializer (url, excludeFragment = false) {
   const href = url.href
   const hashLength = url.hash.length
 
-  return hashLength === 0 ? href : href.substring(0, href.length - hashLength)
+  const serialized = hashLength === 0 ? href : href.substring(0, href.length - hashLength)
```

Comment:
this is a recent bug where serializing `https://localhost/#` would result in `https://localhost/#` instead of `https://localhost/`

Reply:
I dont think endsWith would be slow any longer https://chromium-review.googlesource.com/c/v8/v8/+/4603612 :-)

Reply:
I forgot to add. I meant slower than the way I presented it.

Reply:
I'm not really interested in optimizing against v8, and the link above mostly proves why